### PR TITLE
Daikon logging request interceptor

### PR DESCRIPTION
**Pull request by sdiallo-talend**
**What is the problem this Pull Request is trying to solve?**
  Investigate outbound rest call logging for spring app and incoming/outgoing kafka event logging.
**What is the chosen solution to this problem?**
  Implemente Rest template and kafka interceptor
**Link to the JIRA issue**

 
**Please check if the Pull Request fulfills these requirements**
- [ ] The PR commit message follows our [guidelines](../CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

 
**[ ] This Pull Request introduces a breaking change**

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/83/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/83/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/84/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/84/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/85/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/85/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/daikon_pull-request_master/86/) [:x:](https://ci-common.datapwn.com/job/daikon_pull-request_master/86/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/87/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/87/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/88/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/88/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/daikon_pull-request_master/89/) [:star:](https://ci-common.datapwn.com/job/daikon_pull-request_master/89/)

File: daikon-logging/logging-request-interceptor/src/main/java/org/talend/daikon/logging/TalendKafkaConsumerInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    private static final Logger LOGGER = LoggerFactory.getLogger(TalendKafkaConsumerInterceptor.class);
+
+    @Override
+    public ConsumerRecords<Object, Object> onConsume(ConsumerRecords<Object, Object> records) {
+        if (LOGGER.isTraceEnabled()) {
+            Iterator<ConsumerRecord<Object, Object>> consumerRecords = records.iterator();
+            if (consumerRecords != null) {
+                consumerRecords.forEachRemaining(c -> {
+                    LOGGER.trace(String.format("onConsume topic=%s partition=%d value=%s %d \n", c.topic(), c.partition(),
+                            c.value().toString(), c.partition()));
```
Comment (talend-vlesquere) [inline daikon-logging/logging-request-interceptor/src/main/java/org/talend/daikon/logging/TalendKafkaConsumerInterceptor.java]: c.value() is a serialized message in our product, 
before writing it in the log we need to use KafkaMessageKeyDeserializer which is in org.talend.schema:multi-tenant-kafka

File: daikon-logging/logging-request-interceptor/src/main/java/org/talend/daikon/logging/TalendKafkaProducerInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import org.slf4j.LoggerFactory;
+
+public class TalendKafkaProducerInterceptor implements ProducerInterceptor<Object, Object> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TalendKafkaProducerInterceptor.class);
+
+    @Override
+    public ProducerRecord<Object, Object> onSend(final ProducerRecord<Object, Object> record) {
+        if (LOGGER.isTraceEnabled()) {
+            LOGGER.trace(String.format("onSend topic=%s key=%s value=%s %d \n", record.topic(), record.key(),
```
Comment (talend-vlesquere) [inline daikon-logging/logging-request-interceptor/src/main/java/org/talend/daikon/logging/TalendKafkaProducerInterceptor.java]: record.key and ()record.value() are serialized in our product, 
before writing it in the log we need to use KafkaMessageKeyDeserializer which is in org.talend.schema:multi-tenant-kafka

File: daikon-logging/logging-kafka-interceptor/pom.xml [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+		</dependency>
+		<dependency>
+		    <groupId>org.springframework.kafka</groupId>
+		    <artifactId>spring-kafka</artifactId>
+		    <version>2.1.2.RELEASE</version>
+		    <optional>true</optional>
+		</dependency>
+		<dependency>
+            <groupId>org.talend.daikon</groupId>
+            <artifactId>multitenant-spring-async</artifactId>
```
Comment (sgandon) [inline daikon-logging/logging-kafka-interceptor/pom.xml]: Are you sure you really need this dependency ?

Comment (sdiallo-talend) [reply daikon-logging/logging-kafka-interceptor/pom.xml]: Right, I think I don't need it ))

**Pull request merged**